### PR TITLE
Format datetimes to UTC during serialization only

### PR DESCRIPTION
**Pull request by gdraynz**

### CODE REVIEW

File: nyuki/bus/reporting.py [hunk 1]
```diff
@@ -148,10 +148,9 @@ def send_report(self, rtype, data):
             ),
             'type': rtype,
             'author': self._name,
-            'datetime': datetime.utcnow().isoformat(),
+            'datetime': datetime.utcnow(),
             'data': data
         }
-        self.check_report(report)
```
Comment (gdraynz) [inline nyuki/bus/reporting.py]: That was useless, no need to validate something we just built.

File: nyuki/utils/serialize.py [hunk 2]
```diff
@@ -15,4 +24,4 @@ def _serialize_datetime(dt):
     """
     Datetime serializer.
     """
-    return dt.isoformat()
+    return dt.replace(tzinfo=SimpleUTC()).isoformat()
```
Comment (gdraynz) [inline nyuki/utils/serialize.py]: This is the main change of this issue.
Every datetime object go through this method during serialization.

File: nyuki/bus/reporting.py [hunk 3]
```diff
@@ -148,10 +148,9 @@ def send_report(self, rtype, data):
             ),
             'type': rtype,
             'author': self._name,
-            'datetime': datetime.utcnow().isoformat(),
+            'datetime': datetime.utcnow(),
```
Comment (lhavel) [inline nyuki/bus/reporting.py]: I'd make it even simpler and safer by building a true localized datetime object this way:
```
'datetime': datetime.now(pytz.utc)
```
You'll get something like `datetime.datetime(2017, 5, 3, 12, 32, 3, 989396, tzinfo=<UTC>)` which provides the right string: `dt.isoformat()` returns `"2017-05-03T12:32:03.989396+00:00"`.

File: nyuki/utils/serialize.py [hunk 4]
```diff
@@ -1,5 +1,14 @@
-from datetime import datetime
 from functools import singledispatch
+from datetime import datetime, tzinfo, timedelta
+
+
+class SimpleUTC(tzinfo):
```
Comment (lhavel) [inline nyuki/utils/serialize.py]: See my first comment to avoid subclassing `tzinfo`.

File: nyuki/utils/serialize.py (hunk 2 above)
Comment (lhavel) [inline nyuki/utils/serialize.py]: `dt.isoformat()` is enough with my first comment

File: nyuki/bus/reporting.py (hunk 3 above)
Comment (lhavel) [inline nyuki/bus/reporting.py]:
```
from datetime import datetime, timezone
datetime.now(timezone.utc)
```
does the job as well and avoid using `pytz`

File: nyuki/workflow/workflow.py [hunk 5]
```diff
@@ -273,7 +271,7 @@ def new_workflow(self, template, instance, **kwargs):
             'type': event.data['type'],
             'data': event.data.get('content') or {},
             'source': source,
-            'timestamp': datetime.utcnow().isoformat()
+            'timestamp': utcnow,
```
Comment (lhavel) [inline nyuki/workflow/workflow.py]: `utcnow()`

**Pull request merged**